smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/JMESPathVisitor.kt

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.swift.codegen.waiters
+
+import software.amazon.smithy.jmespath.ExpressionVisitor
+import software.amazon.smithy.jmespath.JmespathExpression
+import software.amazon.smithy.jmespath.RuntimeType
+import software.amazon.smithy.jmespath.ast.AndExpression
+import software.amazon.smithy.jmespath.ast.ComparatorExpression
+import software.amazon.smithy.jmespath.ast.CurrentExpression
+import software.amazon.smithy.jmespath.ast.ExpressionTypeExpression
+import software.amazon.smithy.jmespath.ast.FieldExpression
+import software.amazon.smithy.jmespath.ast.FilterProjectionExpression
+import software.amazon.smithy.jmespath.ast.FlattenExpression
+import software.amazon.smithy.jmespath.ast.FunctionExpression
+import software.amazon.smithy.jmespath.ast.IndexExpression
+import software.amazon.smithy.jmespath.ast.LiteralExpression
+import software.amazon.smithy.jmespath.ast.MultiSelectHashExpression
+import software.amazon.smithy.jmespath.ast.MultiSelectListExpression
+import software.amazon.smithy.jmespath.ast.NotExpression
+import software.amazon.smithy.jmespath.ast.ObjectProjectionExpression
+import software.amazon.smithy.jmespath.ast.OrExpression
+import software.amazon.smithy.jmespath.ast.ProjectionExpression
+import software.amazon.smithy.jmespath.ast.SliceExpression
+import software.amazon.smithy.jmespath.ast.Subexpression
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.utils.toLowerCamelCase
+
+private val suffixSequence = sequenceOf("") + generateSequence(2) { it + 1 }.map(Int::toString) // "", "2", "3", etc.
+
+// Visits the JMESPath syntax tree, rendering the JMESPath expression into a Swift expression.
+// Smithy does not support all JMESPath expressions, and smithy-swift does not support all Smithy
+// expressions.  However, this support is sufficient to generate all waiters currently in use on AWS.
+// Use of a JMESPath feature not supported by Smithy will cause an exception at the time of code
+// generation.
+
+// Because JMESPath has no concept of non-optional, every visitor below will take an optional value
+// as input and also return an optional.
+class JMESPathVisitor(val writer: SwiftWriter) : ExpressionVisitor<String> {
+
+    // A few methods are provided here for generating unique yet still somewhat
+    // descriptive variable names when needed.
+    private val tempVars = mutableSetOf<String>()
+
+    private fun addTempVar(preferredName: String, content: String, vararg args: Any): String {
+        val name = bestTempVarName(preferredName)
+        writer.writeInline("let \$L = ", name)
+        writer.write(content, *args)
+        return name
+    }
+
+    private fun bestTempVarName(preferredName: String): String =
```

Comment:
Best means "unique, based on the preferred name passed in".  So if you ask for `potato` and it hasn't already been used, it will pass back `potato`.  If there is already a `potato` it will pass back `potato2`, then `potato3` on the next call, etc.

Unless you have a better idea, I'll call it `uniqueTempVarName` and add a comment to describe behavior.

Reply:
Nope, and tbh the current name is fine too so defer to w/e you think is best (no pun intended lol). A comment describing that behavior would be great though!